tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -160,15 +159,428 @@ async fn handle_native_task<
-				.await;
+	let Ok(mut rpc_client) = ctx.parentchain_rpc_client_factory.new_client().await else {
+		log::error!("Failed to create rpc client");
+		let response = NativeTaskResponse::Err(NativeTaskError::InternalError);
+		if response_sender.send(response.encode()).is_err() {
+			log::error!("Failed to send response");
+		}
+		return;
+	};
+
+	let auth_type: OmniAccountAuthType = match wrapper.auth {
+		Some(auth) => auth.into(),
+		None => OmniAccountAuthType::NoAuth,
+	};
+
+	let (response_sender, tx) = match wrapper.task {
+		NativeTask::RequestAuthToken(sender) => {
+			let omni_account_storage = MemberOmniAccountStorage::new(ctx.storage_db.clone());
+			let Some(omni_account) = omni_account_storage.get(&sender.hash()) else {
+				let response = NativeTaskResponse::Err(NativeTaskError::UnauthorizedSender);
+				if response_sender.send(response.encode()).is_err() {
+					log::error!("Failed to send response");
+				}
+				return;
+			};
+			let Ok(current_block) = rpc_client.get_last_finalized_block_num().await else {
+				log::error!("Failed to get last finalized block number");
+				let response = NativeTaskResponse::Err(NativeTaskError::InternalError);
+				if response_sender.send(response.encode()).is_err() {
+					log::error!("Failed to send response");
+				}
+				return;
+			};
+			let auth_options = AuthOptions { expires_at: current_block + AUTH_TOKEN_EXPIRATION };
+			let claims = AuthTokenClaims::new(
+				sender.hash().to_string(),
+				AUTH_TOKEN_SESSION_TYPE.to_string(),
+				auth_options,
+			);
+			let Ok(token) = jwt::create(&claims, &ctx.jwt_rsa_private_key) else {
+				let response = NativeTaskResponse::Err(NativeTaskError::AuthTokenCreationFailed);
+				if response_sender.send(response.encode()).is_err() {
+					log::error!("Failed to send response");
+				}
+				return;
+			};
+			let auth_token_requested_call = parentchain_api_interface::tx()
+				.omni_account()
+				.auth_token_requested(AccountId32(omni_account.into()), claims.exp);
+
+			// Without increase nonce, all requests after request_auth_token will failure with below error.
+			// Could not submit tx: Rpc(ClientError(Call(ErrorObject { code: ServerError(1014), message: "Priority is too low: (2564 vs 2564)",
+			// data: Some(RawValue("The transaction has too low priority to replace another transaction already in the pool.")) })))
+			let signer_account_id = ctx.transaction_signer.get_signer_account_id();
+			let nonce = match rpc_client.get_account_nonce(&signer_account_id).await {
+				Ok(n) => n,
+				Err(e) => {
+					log::error!("Failed to get account nonce: {:?}", e);
+					let response = NativeTaskResponse::Err(NativeTaskError::InternalError);
+					if response_sender.send(response.encode()).is_err() {
+						log::error!("Failed to send response");
+					}
+					return;
+				},
+			};
+			// Increment nonce for the next transaction
+			let tx = ctx.transaction_signer.sign(auth_token_requested_call, Some(nonce + 1)).await;
```

Comment:
I don't think we need to increment the nonce here, it can be just `None`. We increment the nonce in the `RequestIntent` case because we submit two extrinsics to the chain, one to track the requested intent and the other to dispatch it (if it's native call) or track the result (if it was on another chain) and the second would fail with the error in the comment above. 

Here we submit one and return the token in the response

Reply:
It's from @wli-pro 's change : )

https://github.com/litentry/heima/pull/3311/files#diff-09ff3956fe238deae5e520798cbb1baee343ddddc01e96c436596d0a48c08472

I also left a comment about inconsistency there

Reply:
I'll create an issue to review these